src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.LockTest.Unix.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Xunit;
+
+public class FileSystemWatcherLockTest
+{
+    [Fact]
+    public static void FileSystemWatcher_WatchLocksWatchedDirectory()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var fsw = new FileSystemWatcher())
+        {
+            fsw.Path = Path.GetFullPath(dir.Path);
+            fsw.EnableRaisingEvents = true;
+
+            Microsoft.Win32.SafeHandles.SafeFileHandle handle = Interop.Sys.Open(fsw.Path, Interop.Sys.OpenFlags.O_RDONLY, 0);
+            Assert.False(handle.IsInvalid); // We should always be able to open a handle
+            Assert.True(Interop.Sys.FLock(handle, Interop.Sys.LockOperations.LOCK_EX | Interop.Sys.LockOperations.LOCK_NB) < 0);
+
+            Interop.ErrorInfo err = Interop.Sys.GetLastErrorInfo();
+            Assert.True(err.Error == Interop.Error.EWOULDBLOCK); // validate it is a sharing violation
+        }
+    }
+}
```

Comment:
Thanks for adding tests.

How did you discover the discrepancy originally?  I'd assumed it was some obvious behavior, like Windows preventing you from creating two FSWs to monitor the same path, but this seems very low-level.   What behaviors does it actually prevent?  Since as you've stated these file locks on Unix are only advisory, and since they don't actually prevent unlinking, and since nothing in our System.IO.FileSystem implementation takes advisory locks on directories, what real behaviors do we expect this change to help with?  I get going for consistency, I'm just wondering if in this case because of the underlying differences if such consistency as actually toothless.